sips/sip-032/sip-032-improved-stacking.md

```diff
@@ -0,0 +1,232 @@
+    section No earning
+    2001 : Prepare phase, no rewards: delegate txs applied only to cycle after next
+    2100 : End of cycle
+```
+
+That means the locking period is 1 cycle, with automatic extension for another cycle until the user decides to end stacking.
+
+The amount of stacking STX can increase during the automatic extension up to the minimum of the balance and a user specified maximum amount.
+
+The amount of stacked STX does not decrease during the automatic extension. If the user requested a decrease of the stacked amount, this difference is unlocked at the end of the stacking period.
+
+## Semantic Change of Delegation
+
+When signers verify and accept proposed blocks by miners, their voting power corresponds to the amount of stacked Stacks tokens (see SIP-021). The new stacking process changes the delegation flow as follows:
+
+- `delegate-stx` shall be renamed to `designate`. The function takes the arguments: `amount`, `signature`, optional bitcoin `block-height` defining the end of stacking and auto extending, optional `pox-addr` defining that pox-address that the signer must use, optional `max-amount`. The `amount` defines how many Stacks tokens are locked immediately. The `max-amount` defines the maximum of Stacks tokens that can be locked in the future through auto extending. If provided the minimum of the user's stx balance and `max-amount` is locked. If omitted, the whole user's balance is locked. Users can use sponsored transactions to call revoke-delegate-stx in case there is no unlocked stx in the account. The argument `signature` is a signature of the signer indicating that the signer accepted the delegation of voting power. The signing follows the structured message signing with the parameters and topic `designate` as message. The public key of the signature can be used to identify the signer similar to the Stacks address of the pool operator in the previous PoX design.
+
+- `revoke-delegate-stx` shall be renamed to `revoke-designation`. After calling this function, auto extension is stopped and Stacks tokens are unlocked for the user at the end of the current cycle. If a signer does not aggregate enough Stacks tokens to receive at least one reward slot (or does not commit at all) then the user's Stacks token are unlocked immediately through this call.
+
+- `delegate-increase` is replaced by `update-max-amount`. It sets max-amount to the new value. This value is applied when the user's designation is extended for the next cycle. The amount can be smaller than the currently locked amount. As tokens are only locked for 1 cycle at a time, handling decreasing is now easy enough in comparison to the previous system with locking periods of up to 12 cycles.
+
+- `delegate-extend` is removed because the locking period is automatically extended each cycle.
+
+## Amount of Stacked STX
+
+When a transaction locks or unlocks STX for the user a corresponding event shall be emitted. The following events used in PoX 4 are used:
```

Comment:
PoX 4 uses the function's name as the event's name (i.e. in the event tuple's `name` field).  Is this mapping to PoX 5 correct?

* `delegate-stx` --> `designate`
* `designate-increase` --> `stacks-aggregation-increase`
* `designate-decrease` --> (this is new, and has no PoX 4 analogue)
* `designate-extend` --> similar to `delegate-stx-extend`, but is now automatic?
* `signer-accept` --> `stacks-aggregation-commit`

Can you also explain what the event payloads will be?

Reply:
Ah, I see that `stacks-aggregate-commit` is due to be replaced with `signer-register`.  Is `signer-accept` a new function in PoX 5?  I don't see it defined in this document.